src/crepe/system/InputSystem.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include "System.h"
+#include "../facade/SDLContext.h"
+
+namespace crepe {
+
+class Button;
+class Transform;
+/**
+ * \class InputSystem
+ * \brief Handles the processing of input events like mouse and keyboard interactions.
+ * 
+ * This system processes events such as mouse clicks, mouse movement, and keyboard
+ * actions. It is responsible for detecting interactions with UI buttons and
+ * passing the corresponding events to the registered listeners.
+ */
+class InputSystem : public System {
+public:
+	using System::System;
+
+	/**
+     * \brief Updates the system, processing all input events.
+     * This method processes all events and triggers corresponding actions.
+     */
+	void update() override;
+
+private:
+	//! Stores the last position of the mouse when the button was pressed.
+	std::pair<int, int> last_mouse_down_position{-1, -1};
+
+	//! Stores the last mouse button pressed.
+	MouseButton last_mouse_button = MouseButton::NONE;
+
+	//! The tolerance in pixels for detecting a mouse click.
+	const int click_tolerance = 5;
```

Comment:
Dit kan beter een waarde in game units worden. We werken immers niet met pixels in onze game engine.